lgc/builder/DescBuilder.cpp

```diff
@@ -411,55 +430,66 @@ Value *DescBuilder::getDescPtr(ResourceNodeType resType, unsigned descSet, unsig
                                const ResourceNode *topNode, const ResourceNode *node, bool shadow) {
   Value *descPtr = nullptr;
 
-  // Get the descriptor table pointer.
-  // TODO Shader compilation: If we do not have user data layout info (topNode and node are nullptr), then
-  // we do not know at compile time whether a DescriptorBuffer is in the root table or the table for its
-  // descriptor set, so we need to generate a select between the two, where the condition is a reloc.
-  if (node && node == topNode) {
+  static unsigned i = 0;
+  llvm::errs() << "getDescPtr " << ++i << ", resType: " << resType << ", node: " << node << "\n";
+
+  auto GetSpillTableAddr = [this]() -> Value * {
```

Comment:
1. camelBack
2. You don't need to specify the return type if the compiler can find it out from the return statement, right?
3. I'll defer to others as I'm a bit of a luddite when it comes to C++ features, but I'm not sure about using lambdas here. I think I would have written it with one "if" to get the spill table, another "if" to get the descriptor table, and a third "if" to do the select with the reloc.

Reply:
Agreed.

Reply:
Fixed 1 and 2. RE 3, I think that in this particular case the two lambdas make the code more readable. It's somewhat awkward to structure the code like you suggested, as you have to reorder the if conditions to make the relocatable case come last. I also like that the lambdas can provide descriptive naming for code snippets that make sense in some specific context, which is easier to keep up to that than commends, but I can see that you could get the same benefit here using local variables for generating the addresses.